### DISCUSSION TITLE
ERROR File download mis-match issue

### DISCUSSION THREAD

**Discussion by PSUHammer** [abraunegg/onedrive#2581] [Q&A] [answered]
I am only performing a download sync locally for a few folders to a local Ubuntu Plex server.  I get this error on a fresh setup:

```
Jan 07 16:16:10 PC onedrive[9011]: [3.5K blob data]
Jan 07 16:16:10 PC onedrive[9011]: ERROR: File download size mis-match. Increase logging verbosity to determine why.
Jan 07 16:16:10 PC onedrive[9011]: ERROR: File download hash mis-match. Increase logging verbosity to determine why.
Jan 07 16:16:10 PConedrive[9011]: INFO: Potentially add --disable-download-validation to work around this issue but downloaded data integrity cannot be guaranteed.
Jan 07 16:16:10 PC onedrive[9011]: Downloading file Pictures/2023/July_2023_071723_59.heic ...
```

I have no idea how to troubleshoot this.  I would assume it should just pull and dump the files.  Doesn't seem to stop it and it continues on.

**Comment by abraunegg** [comment, marked as answer]
@PSUHammer 
> I have no idea how to troubleshoot this.

Please read the log .. it is advising you what to do .... `ERROR: File download size mis-match. Increase logging verbosity to determine why.`

Please read the documentation on how to increase your logging level.

Secondly - the next line you are providing is this:
```
Jan 07 16:16:10 PC onedrive[9011]: Downloading file Pictures/2023/July_2023_071723_59.heic ...
```

There is a *known* Microsoft API **bug** that is causing you **data loss**. You should be highly concerned regarding the integrity of your data. Please read: 
* https://github.com/abraunegg/onedrive/issues/2471
* https://github.com/OneDrive/onedrive-api-docs/issues/1532
* https://github.com/OneDrive/onedrive-api-docs/issues/1723

**Reply by PSUHammer** [reply]
Sorry man, I did read the log and understand I can ignore data integrity, which I will do, but was just checking on if this was a known issue.   I also enabled DEBUG verbosity for logging but it's not overly helpful.

Thanks for pointing me to the API bug issue.  That may be part of the reason that OnedriveGUI wasn't working as well.  Looks like the culprit.

I am not highly worried as it is a one way download sync for photos/videos for my plex server.

**Reply by abraunegg** [reply]
You actually should be worried - as what you are downloading is not the same as what is stored online ....

**Reply by PSUHammer** [reply]
I mean, it sucks, but I have multiple copies of these photos.  I am using this app just to get a quick and easy copy to my Plex server.  Hopefully it gets straightened out by MS at some point.

To be honest, I am more concerned about Plex not supporting the HEIC format yet.
